software/README.md

```diff
@@ -49,25 +49,34 @@ If you really want to avoid 'noise' which would present as a flickering value de
 The ADCs used to read the knob position are only 12 bit, which means that any read_position value above 4096 (2^12) will not actually be any finer resolution, but will instead just go up in steps. For example using 8192 would only return values which go up in steps of 2.
 
 
-## Digital Input and Buttons
+## Digital Input
 
-It may seem a little strange to see the digital input and buttons grouped together, but if you think about it, all the button is doing is create an internal digital input when you press it. For this reason, the method for handling the event that should happen when either a button is pressed or a digital input detected is the same.
-
-A class is used named 'DigitalInput' in the europi.py library, which handles both the debounce (only used for buttons) and the handler (what happens when a press/input is detected).
+The Digital Input jack can detect a HIGH signal when recieving voltage > 0.8v and will be LOW when below.
 
 | Method        | Usage       | Parameter(s)       |
 | ------------- | ----------- | ----------- |
 |value|Reads the current value of the input, HIGH (1) or LOW (0).|
-|handler|Assign a new function to be used as the handler|function
+|handler|Define the callback function to call when a rising edge is detected|function
+|handler_falling|Define the callback function to call when a falling edge is detected|function
 |reset_handler|Detach the handler method from the Pin IRQ|
-
+|last_triggered|Return the duration in milliseconds from the start of the last trigger.|
 
 To use the handler method, you simply define whatever you want to happen when a button or the digital input is triggered, and then use x.handler(new_function). Do not include the brackets for the function, and replace the 'x' in the example with the name of your input, either b1, b2, or din.
 
-Button instances have an attribute `last_pressed` which can be used to perform some action or behavior relative to when the button was last pressed. For example, if you want to display that a button was pressed, you could add the following code to your main script loop:
+## Button
+
+| Method        | Usage       | Parameter(s)       |
+| ------------- | ----------- | ----------- |
+|value|Reads the current value of the input, HIGH (1) or LOW (0).|
+|handler|Define the callback function to call when the button is pressed|function
+|handler_falling|Define the callback function to call when the button is released|function
+|reset_handler|Detach the handler method from the Pin IRQ|
+|last_pressed|Return the duration in milliseconds from when the button was last pressed.|
```

Comment:
This line might be simpler as 'return the duration in milliseconds *since* the last button press*
There's a potential for confusion in the current wording where it could be referring to the actual length of time the button was held down for

Reply:
Good call, thanks for helping improve the clarity. I have fixed this in PR #62 where it was originally introduced.